### PR TITLE
feat(coinmarket): best rate selecting

### PR DESCRIPTION
**Pull request by adderpositive**
## Info
A tag `Best rate` is added only to offers with the highest amount the client receives

### CODE REVIEW

File: packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts [hunk 1]
```diff
@@ -273,3 +276,24 @@ export function processSellAndBuyQuotes<T extends CoinmarketTradeBuySellType>(
...
+    const getTypedAmount = (quote: CoinmarketTradeDetailType) => {
+        if (type === 'sell') {
+            return new BigNumber((quote as SellFiatTrade).fiatStringAmount ?? 0);
+        }
+
+        return new BigNumber((quote as BuyTrade | ExchangeTrade).receiveStringAmount ?? 0);
+    };
+
+    const bestRatedQuotes = quotes
+        ? quotes.toSorted((a, b) => getTypedAmount(b).minus(getTypedAmount(a)).toNumber())
```
Comment (FreeWall) [inline packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts]: is `toSorted` supported in all browsers suite supports?

Comment (adderpositive) [reply packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts]: `sort` will be safer! Thanks

File: packages/suite/src/hooks/wallet/coinmarket/offers/useCoinmarketSellOffers.ts [hunk 2]
```diff
@@ -82,7 +83,9 @@ export const useCoinmarketSellOffers = ({
 
                 setInnerQuotes(
                     getFilteredSuccessQuotes<CoinmarketTradeSellType>(
-                        processSellAndBuyQuotes<CoinmarketTradeSellType>(allQuotes),
+                        filterQuotesAccordingTags<CoinmarketTradeSellType>(
+                            addIdsToQuotes<CoinmarketTradeSellType>(allQuotes, 'sell'),
+                        ),
                     ),
                 );
```
Comment (LukasRada) [inline packages/suite/src/hooks/wallet/coinmarket/offers/useCoinmarketSellOffers.ts]: Please extract function result into const variables and use them accordingly.

File: packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts [hunk 3]
```diff
@@ -257,19 +261,51 @@ export const getDefaultCountry = (country: string = regional.unknownCountry) =>
...
+              }
+
+              // descending to rate for sell/exchange - higher rate more crypto/fiat client receives
+              return new BigNumber(b.rate ?? 0).minus(new BigNumber(a.rate ?? 0)).toNumber();
+          })
+        : null;
+    const bestRatedQuote = bestRatedQuotes?.[0];
+
+    return bestRatedQuote;
+};
```
Comment (LukasRada) [inline packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts:327]: Maybe move the logic to server? cc: @FreeWall

**Review by LukasRada — CHANGES_REQUESTED** [review]

File: packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts (hunk 3 above)
Comment (FreeWall) [reply packages/suite/src/utils/wallet/coinmarket/coinmarketUtils.ts:327]: Best rated quote (or `quotes` array in this case) depends on client-side filtering (payment method), so it's calculated on client-side.
In order to use logic on server side, we would need to calculate best rate for each payment method and pass the info to client somehow.

**Pull request merged**